src/app/routes/verifiable-presentation/verifiable-presentation/verifiable-presentation.component.ts

```diff
@@ -1,16 +1,42 @@
 import { Component, OnInit } from '@angular/core';
-import { MatTableDataSource } from '@angular/material/table';
+import { ActivatedRoute } from '@angular/router';
+import { Observable } from 'rxjs';
+import { IamService } from "../../../shared/services/iam.service";
+import { IOOBPresentation } from '../models/out-of-band-presentation.interface';
+import { LoadingService } from '../../../shared/services/loading.service';
 
 @Component({
-  selector: 'app-verifiable-presentation',
-  templateUrl: './verifiable-presentation.component.html',
-  styleUrls: ['./verifiable-presentation.component.scss']
+    selector: 'app-verifiable-presentation',
+    templateUrl: './verifiable-presentation.component.html',
+    styleUrls: ['./verifiable-presentation.component.scss']
 })
 export class VerifiablePresentationComponent implements OnInit {
+    oob$: Observable<string>
 
-  constructor() { }
-
-  ngOnInit(): void {
-  }
-
+    constructor(
+        private route: ActivatedRoute,
+        private loadingService: LoadingService
+    ) { }
+    isAutolistLoading = {
+        requests: [],
+        value: false,
+      };
+        //logic to decode the presentation
+        ngOnInit(): void {
+            this._initSearch()
+    
+        }
+ 
+    private _initSearch() {
+        this.route.queryParams.subscribe(async (params: any) => {
+            this.loadingService.show()
```

Comment:
If there is if statement inside observable (without else) then i prefer just to filter it out. Parsing you can move to `map` operator (first you need to use `pipe()` method of observable). if you are afraid of debugging it, then there is an operator `tap` that do not change object, but you can check what is going on inside or do other stuff like displaying loader :) For hiding i'm using `finalize` operator, which works like `finally` in try-catch block.